### ISSUE TITLE
`described_class` is overridden in nested example group

### ISSUE THREAD

**Issue by pirj** [rspec/rspec-core#2627]
### Subject of the issue

The following spec file:

```
module A; end
module B; end

RSpec.describe A do
  describe B do
    it { expect(described_class.name).to eq('B') }
  end
end
```

passes.

[`described_class` docs](https://relishapp.com/rspec/rspec-core/v/3-8/docs/metadata/described-class) mention:

> If the first argument to the **outermost** example group is a class, the class is
exposed to each example via the described_class() method.

### Your environment
* Ruby version: 2.5.5
* rspec-core version: 3.8.2

### Steps to reproduce

Run the spec from description.

### Expected behavior

Example fails:
```
Failures:

  1) A B should eq "B"
     Failure/Error: it { expect(described_class.name).to eq('B') }

       expected: "B"
            got: "A"
```
### Actual behavior

Example is green.

**Comment by JonRowe** [comment]
This is deliberate, see b39385e6624d2f42f588eda7b82a68cf6d867eb7#diff-a0aa116f3620e98f10c6a6bf9586e1c2 from #1114 and #1361.

So at worst the docs are incomplete here!

**Comment by pirj** [comment]
Thanks for the clarification, @JonRowe, this behaviour makes sense to me.

One [`rubocop-rspec` cop](https://github.com/rubocop-hq/rubocop-rspec/blob/24ad67b/spec/rubocop/cop/rspec/described_class_spec.rb#L140) is off, as it was relying on the outdated documentation, I stumbled across this mismatch.

Fixed the doc in #2629
